src/main/java/uk/gov/hmcts/dts/fact/model/Court.java

```diff
@@ -87,7 +90,11 @@ public Court(uk.gov.hmcts.dts.fact.entity.Court courtEntity) {
             .map(Contact::new).collect(toList());
         this.courtTypes = courtEntity.getCourtTypes().stream().map(CourtType::getName).collect(toList());
         this.emails = courtEntity.getEmails().stream().map(Email::new).collect(toList());
-        this.openingTimes = courtEntity.getOpeningTimes().stream().map(OpeningTime::new).collect(toList());
+        this.openingTimes = ofNullable(courtEntity.getCourtOpeningTimes())
+            .map(Collection::stream)
+            .orElseGet(Stream::empty)
```

Comment:
Do you need this? I always thought map of an optional just did nothing. I guess you would need to put an orElse at the end with an empty list so it probably amounts to the same thing.

Reply:
yea, it protects against null, replacing it with an empty stream